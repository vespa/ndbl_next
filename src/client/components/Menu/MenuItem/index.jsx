import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import cssMain from '../../../../scss/main.scss';

const MenuItem = ({ href, text }) => {
  return (
    <Link href={href}>
      <a href={href} className={`${cssMain.col_3}`}>
        {text}
      </a>
    </Link>
  );
};

MenuItem.propTypes = {
  href: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default MenuItem;
