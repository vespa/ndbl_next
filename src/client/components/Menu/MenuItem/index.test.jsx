import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from './index';

describe('[Component: Input]', () => {
  let wrapper;
  const props = {
    text: 'Name',
    href: 'name',
  };
  beforeEach(() => {
    wrapper = shallow(<MenuItem {...props} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
