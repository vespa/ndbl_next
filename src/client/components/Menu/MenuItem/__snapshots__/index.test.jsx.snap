// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: MenuItem] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    href="name"
    text="Name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <a
        className="undefined"
        href="name"
      >
        Name
      </a>,
      "href": "name",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Name",
        "className": "undefined",
        "href": "name",
      },
      "ref": null,
      "rendered": "Name",
      "type": "a",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <a
          className="undefined"
          href="name"
        >
          Name
        </a>,
        "href": "name",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
          "className": "undefined",
          "href": "name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "a",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
